compiler/tools/suggest.nim

```diff
@@ -466,6 +468,48 @@ proc isTracked*(current, trackPos: TLineInfo, tokenLen: int): bool =
     if col >= current.col and col <= current.col+tokenLen-1:
       return true
 
+proc findTrackedNode(n: PNode; trackPos: TLineInfo): PSym =
+  if n.kind == nkSym:
+    if isTracked(n.info, trackPos, n.sym.name.s.len): return n.sym
+  else:
+    for i in 0 ..< safeLen(n):
+      let res = findTrackedNode(n[i], trackPos)
+      if res != nil: return res
+
+proc findTrackedSym*(g: ModuleGraph;): PSym =
+  let m = g.getModule(g.config.m.trackPos.fileIndex)
+  if m != nil and m.ast != nil:
+    result = findTrackedNode(m.ast, g.config.m.trackPos)
+
+proc executeCmd*(cmd: IdeCmd, file, dirtyfile: AbsoluteFile, line, col: int;
```

Comment:
public procedures of modules, especially important ones like these should have a doc comment describing them.

Reply:
could you help me with this?

Reply:
Something like:

> executes the given suggest command, `cmd`, for a given `file`, at the position described by `line` and `col`umn. If `dirtyFile` is non-empty, then its contents are used as part of the analysis.

Something like that, not sure if I'm missing key bits of functionality?

Reply:
A doc comment for a routine should cover what it does and any pre/post conditions about the routine/its inputs.